docs/contributor_guide/development_process.rst

```diff
@@ -14,6 +14,18 @@ run periodically. When adding new code paths or features, tests are a
 requirement to complete a pull request. They should be added in the
 :code:`test` directory.
 
+Documentation should be provided with pull requests that add or change
```

Comment:
Do we want to mention CHANGES.md and AUTHORS.md

Reply:
Good point